services/wallet/fees.go

```diff
@@ -163,6 +176,57 @@ func (f *FeeManager) suggestedFees(ctx context.Context, chainID uint64) (*Sugges
 	}, nil
 }
 
+func (f *FeeManager) getBaseFee(ctx context.Context, client chain.ClientInterface, testnetMode bool) (*big.Int, error) {
+	header, err := client.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	config := params.MainnetChainConfig
+	if testnetMode {
+		config = params.SepoliaChainConfig
+	}
+
+	baseFee := misc.CalcBaseFee(config, header)
+	return baseFee, nil
+}
+
+func (f *FeeManager) getPriorityFees(ctx context.Context, client chain.ClientInterface, baseFee *big.Int) (PriorityFees, error) {
+	var priorityFee PriorityFees
+	fees, err := f.getFeeHistorySorted(client.NetworkID())
+	if err != nil {
+		return priorityFee, err
+	}
+
+	suggestedPriorityFee, err := client.SuggestGasTipCap(ctx)
+	if err != nil {
+		return priorityFee, err
+	}
+
+	// Calculate Low priority fee
+	priorityFee.Low = fees[int64(0.1*float64(len(fees)))-1]
```

Comment:
How does counting the number of fees help determine the fee value?

`fees, err := f.getFeeHistorySorted(client.NetworkID())` gets historic fees sorted, but `len(fees)` just counts the number of fees, it doesn't handle the fee values.

Reply:
This logic was taken from `suggestedFees` function. I guess it's about 10th percentile, so we sort fees in an ascending way and take one that 10% of fees are under it, as the the low priority fee.
But if I'm mistaken @alaibe can explain better.

Reply:
Oh, I see. Yes, thank you.

Reply:
correct it is 10th percentile